lib/pltr/v2/selectors/__tests__/categories.test.js

```diff
@@ -0,0 +1,162 @@
+import { addCharacter, editCategory } from '../../actions/characters'
+import { addNoteWithValues, editNote } from '../../actions/notes'
+import { changeCurrentView, loadFile } from '../../actions/ui'
+import { emptyFile } from '../../store/newFileState'
+import {
+  noteCategoriesSelector,
+  sortedCharacterCategoriesSelector,
+  sortedNoteCategoriesSelector,
+} from '../categoriesFirstOrder'
+import { categoriesFilterItemsSelector } from '../categoriesThirdOrder'
+import { charactersSortedInBookSelector } from '../charactersThirdOrder'
+import { configureStore } from './fixtures/testStore'
+
+const EMPTY_FILE = emptyFile('Test file')
+const initialStore = () => {
+  const store = configureStore()
+  store.dispatch(
+    loadFile(
+      'Test file',
+      false,
+      EMPTY_FILE,
+      EMPTY_FILE.file.version,
+      'device://tmp/dummy-url-test-file.pltr'
+    )
+  )
+  return store
+}
+
+const uncategorized = {
+  id: null,
+  name: 'Uncategorized',
+  position: -1,
+  type: 'text',
+}
+
+describe('categoriesFilterItemsSelector', () => {
+  const store = initialStore()
+  const initialState = store.getState().present
+  describe('given a new file', () => {
+    it('should produce the empty array', () => {
+      expect(categoriesFilterItemsSelector(initialState)).toEqual([])
+    })
+  })
+
+  describe('move to tab with categories on filter (Notes)', () => {
+    store.dispatch(changeCurrentView('notes'))
+    const stateAfterChangingView = store.getState().present
+    describe('given the view has no contents', () => {
+      it('should produce the empty array', () => {
+        expect(categoriesFilterItemsSelector(stateAfterChangingView)).toEqual([])
+      })
+    })
+    describe('add notes', () => {
+      store.dispatch(addNoteWithValues('note 1', 'note 1 description'))
+      const stateAfterAddingNote = store.getState().present
+      it('should have 1 filter item', () => {
+        const currentFilterItems = categoriesFilterItemsSelector(stateAfterAddingNote)
+        expect(currentFilterItems.length).toBe(1)
+      })
+      it('should have the uncategorized filter item', () => {
+        const currentFilterItems = categoriesFilterItemsSelector(stateAfterAddingNote)
+        expect(currentFilterItems).toEqual([uncategorized])
+      })
+
+      describe('change category', () => {
```

Comment:
Is this the correct behaviour?  (We won't be able to filter to uncategorised characters.)

Reply:
NVM.  I was wrong.